File: src/com/bulletphysics/collision/broadphase/DbvtProxy.java
package com.bulletphysics.collision.broadphase;

/**
 * Dbvt implementation by Nathanael Presson
 * @author jezek2
 */
public class DbvtProxy extends BroadphaseProxy {

    public final DbvtAabbMm aabb = new DbvtAabbMm();
    public DbvtNode leaf;
    public final DbvtProxy[] links = new DbvtProxy[2];
    public int stage;

    public DbvtProxy(Object userPtr, short collisionFilterGroup, short collisionFilterMask) {
        super(userPtr, collisionFilterGroup, collisionFilterMask);
    }

}

File: src/com/bulletphysics/collision/broadphase/DbvtProxy.kt
package com.bulletphysics.collision.broadphase

/**
 * Dbvt implementation by Nathanael Presson
 * @author jezek2
 */
class DbvtProxy(userPtr: Any?, collisionFilterGroup: Short, collisionFilterMask: Short) :
    BroadphaseProxy(userPtr, collisionFilterGroup, collisionFilterMask) {
    val aabb: DbvtAabbMm = DbvtAabbMm()
    var leaf: DbvtNode? = null
    val links: Array<DbvtProxy?> = arrayOfNulls<DbvtProxy>(2)
    var stage: Int = 0
}
